File: src/test/kotlin/notjoe/stockpile/testutil/TestItems.kt
package notjoe.stockpile.testutil

import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.util.IItemProvider

class TestItem(val name: String, builder: Builder? = Item.Builder()) : Item(builder) {
    override fun makeUnlocalizedName(): String {
        return "test.$name"
    }

    override fun toString(): String {
        return "[TEST] $name"
    }
}

val RED_ITEM: Item = TestItem("red")
val BLUE_ITEM: Item = TestItem("blue")

infix fun Int.of(item: IItemProvider) = ItemStack(item, this)

File: src/test/scala/notjoe/stockpile/test/TestItems.scala
package notjoe.stockpile.test

import net.minecraft.item.{Item, Items}
import org.powermock.reflect.Whitebox

object TestItems {
  // We need to make sure AIR is set so methods like ItemStack::isEmpty work properly
  Whitebox.setInternalState(classOf[Items], "AIR", new Item(new Item.Settings()).asInstanceOf[Any])

  val Red = new Item(new Item.Settings())
  val Blue = new Item(new Item.Settings())
}
